### DISCUSSION TITLE
Unsupported BMW Models

### DISCUSSION THREAD

**Discussion by my-ix** [BimmerGestalt/AAIdrive#316] [General]
Hi, sorry it this is found out of topic, I want to try out your application, but when I read the documentation, it says this app will use the BMW Connected installed on our phone as proxy. My problem is that the car that I am driving is earlier than 2014, it's 2013 F10 520d. After I entered the VIN number on the app, it said vehicle not supported. I have updated my head unit firmware, but later found out that, the app actually verify WITH BMW server to check. The question is, can you app work without BMW Connected? and is there a way to bypass the VIN verification to straight away get it registered?

**Comment by hufman** [comment]
A very good question! Indeed, the app itself only talks over TCP to the proxy run by BMW Connected or MyBMW, and does not have the code to talk to the car directly.
However, BMW Connected (as far as I can tell) does not talk to the server to confirm whether a specific car can run the Apps protocol. I suspect this VIN entry is used for registering the car with your ConnectedDrive profile online and showing car information in the app, but the actual Car App proxy itself runs automatically in the background for any compatible car (unless Connected 6.5 has received the signal to shut down). Experiments have found that BMW Connected 6.4 doesn't even need to be signed in, and will continue to run the Apps protocol behind the login screen.
In contrast, it seems that MyBMW does check a car's compatibility online, and if the car isn't in the app (my American car is omitted in the European MyBMW app, for example) it will continuously try the new IDrive7 connection instead of trying the legacy Apps protocol.

As for the compatibility of your car, the big thing is that it needs to implement the Android Open Accessory protocol over USB. When you plug in the USB connection to a data port (and make sure it isn't set to Charge Only mode) and the car implements this protocol, it should show a popup saying something like "Open Connected to handle this accessory?". If this popup doesn't happen, then your car probably doesn't implement the Android interface for the Apps protocol.

Do you have the CIC generation of IDrive? (See previous musings at #241) I believe CIC supports BMW Apps from iPhone, and I believe the App protocol itself is roughly the same. Instead of a serial connection within the Android Open Accessory protocol, it's instead wrapped inside the iPhone Accessory Protocol. However, Android itself can't talk this protocol, so there'd need to be some converter box in the line, with someone skilled in USB hackery to build that device.

From what I've found, your 2014 F10 should have NBT, and so it _should_ just work with Android. There might be some coding things to do, but NBT should be capable of it. What does your phone say when you plug in the cable to the car?

**Reply by my-ix** [reply]
On the car screen, in Connected Drive it asks to download BMW Connected apps, even I already have it.

What I just did just now is by downloading BMW Connected from APKPure, and it can now works

Looks like the one that I got from Play Store connect to different server compared to the one from APKPure that's refers to US. I re-register my account to make it work.

Now I can run your application.

Just to ask, in my testing, I tried BMW Connected Legacy, which actually works, even though it's old; this created two applications in my car Connected Drive, one of them is Calendar that not working, any idea to reset this so that it can be removed?

**Reply by hufman** [reply]
I'm glad you got it working! Make sure to disable Auto-Update so the Play Store doesn't upgrade it to Connected 6.5.
I think those leftover icons can be cleared up by simply wiping the storage for the Connected app.

**Reply by my-ix** [reply]
Actually those apps was created by the BMW Connected Legacy. I tried to clear storage and cache on my phone app, but it doesn't work, I still have BMW Calendar and BMW Connected app left. Even though BMW Connected version 6.4 works, but I still can see notification on my ConnectedDrive - Connection Assistance : "App not compatible with the vehicle. Update the app if necessary". I hope the app that you are building can become standalone to communicate natively without using BMW Connected app as proxy. I can become your tester or to assist where possible.

**Reply by my-ix** [reply]
@hufman any plan to make your app native? looks like BMW will retire the connectedDrive replacing it with MyBMW, the sad part is that, only 6.4 and below version support those car older than 2014. Hope you can find a way to.

**Reply by bogdan-calapod** [reply]
@my-ix you can try the latest build with MyBMW - I've been running it for a couple of weeks and it works just fine :D

**Comment by my-ix** [comment]
I have tried, but my connectedDrive not able to connect to the app via USB. My car is 2013 so not supported. Only those 2014 and recent able to.

**Reply by hufman** [reply]
There's been some recent news, where it seems that MyBMW is able to sometimes connect over USB: A user discovered a reliable connection sequence [over in this thread](https://github.com/hufman/AndroidAutoIdrive/discussions/238#discussioncomment-935381).
Granted, MyBMW does seem to apply a little more stricter logic when deciding what apps to connect over Bluetooth (I think it looks up the car model on your ConnectedDrive account to know whether to try to connect the Apps or with Android Auto), and so it might be more reliable to continue using BMW Connected 6.4.
Instead of reimplementing the protocol at that layer, I'd lean towards recommending people obtain the older official app.
